components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/spark/algorithms/StackedAutoencodersClassifier.java

```diff
@@ -0,0 +1,218 @@
+                // This is causin all the predictions to be 0.0
+                // p._autoencoder = true;
+                deeplearningParameters._activation = getActivationType(activationType);
+                deeplearningParameters._hidden = layerSizes;
+                deeplearningParameters._train_samples_per_iteration = batchSize;
+                deeplearningParameters._input_dropout_ratio = 0.2;
+                deeplearningParameters._l1 = 1e-5;
+                deeplearningParameters._max_w2 = 10;
+                deeplearningParameters._epochs = epochs;
+
+                // speed up training
+                deeplearningParameters._adaptive_rate = true; // disable adaptive per-weight learning rate -> default
+                                                              // settings for learning rate and momentum are probably
+                                                              // not ideal (slow convergence)
+                deeplearningParameters._replicate_training_data = true; // avoid extra communication cost upfront, got
+                                                                        // enough data on each node for load balancing
+                deeplearningParameters._overwrite_with_best_model = true; // no need to keep the best model around
+                deeplearningParameters._diagnostics = false; // no need to compute statistics during training
+                deeplearningParameters._classification_stop = -1;
+                deeplearningParameters._score_interval = 60; // score and print progress report (only) every 20 seconds
+                deeplearningParameters._score_training_samples = batchSize / 10; // only score on a small sample of the
+                                                                                 // training set -> don't want to spend
+                                                                                 // too much time scoring (note: there
+                                                                                 // will be at least 1 row per chunk)
+
+                DKV.put(trainFrame);
+                DKV.put(vframe);
+
+                deeplearning = new DeepLearning(deeplearningParameters);
+
+                log.info("Start training deeplearning model ....");
+                try {
+                    dlModel = deeplearning.trainModel().get();
+                    log.info("Successfully finished Training deeplearning model ....");
+                } catch (RuntimeException ex) {
+                    log.info("Error in training Stacked Autoencoder classifier model");
+                    log.info(ex.getMessage());
+                }
+            } else {
+                log.error("Train file not found!");
+            }
+        } catch (RuntimeException ex) {
+            log.info("Failed to train the deeplearning model [id] " + modelID + ". " + ex.getMessage());
```

Comment:
Method `trainModel` doesn't throw any exception and the runtime exception is not known